examples/dynamicrafter/lvdm/modules/networks/openaimodel3d.py

```diff
@@ -0,0 +1,822 @@
+    :param use_conv: if True and out_channels is specified, use a spatial
+        convolution instead of a smaller 1x1 convolution to change the
+        channels in the skip connection.
+    :param dims: determines if the signal is 1D, 2D, or 3D.
+    :param use_checkpoint: if True, use gradient checkpointing on this module.
+    :param up: if True, use this block for upsampling.
+    :param down: if True, use this block for downsampling.
+    """
+
+    def __init__(
+        self,
+        channels,
+        emb_channels,
+        dropout=1.0,
+        out_channels=None,
+        use_conv=False,
+        use_scale_shift_norm=False,
+        dims=2,
+        use_checkpoint=False,
+        up=False,
+        down=False,
+        dtype=ms.float32,
+        use_temporal_conv=False,
+        tempspatial_aware=False
+        # norm_in_5d=False,
+    ):
+        super().__init__()
+        self.channels = channels
+        self.emb_channels = emb_channels
+        self.dropout = dropout
+        self.out_channels = out_channels or channels
+        self.ori_channels = channels
+        self.use_conv = use_conv
+        self.use_checkpoint = use_checkpoint
+        self.use_scale_shift_norm = use_scale_shift_norm
+        self.updown = up or down
+        self.dtype = dtype
+        self.use_temporal_conv = use_temporal_conv
+        self.identity = Identity()
+        self.split = ops.Split(1, 2)
+
+        # self.in_layers_norm = normalization(
+        #     channels, norm_in_5d=norm_in_5d
+        # )  # TODO: this is group norm actually, wrong naming. but renaming requires update of ckpt param name or mapping dict.
+        self.in_layers_norm = normalization(channels)  # TODO: this is group norm actually, wrong naming. but renaming requires update of ckpt param name or mapping dict.
+        self.in_layers_silu = nn.SiLU().to_float(self.dtype)
+        self.in_layers_conv = conv_nd(
+            dims, channels, self.out_channels, 3, padding=1, has_bias=True, pad_mode="pad"
+        ).to_float(self.dtype)
+
+        if up:
+            self.h_upd = Upsample(channels, False, dims, dtype=self.dtype)
+            self.x_upd = Upsample(channels, False, dims, dtype=self.dtype)
+        elif down:
+            self.h_upd = Downsample(channels, False, dims, dtype=self.dtype)
+            self.x_upd = Downsample(channels, False, dims, dtype=self.dtype)
+        else:
+            self.h_upd = self.x_upd = self.identity
+
+        self.emb_layers = nn.SequentialCell(
+            nn.SiLU().to_float(self.dtype),
+            linear(
+                emb_channels, 2 * self.out_channels if use_scale_shift_norm else self.out_channels, dtype=self.dtype
+            ),
+        )
+
+        # self.out_layers_norm = normalization(self.out_channels, norm_in_5d=norm_in_5d)
+        self.out_layers_norm = normalization(self.out_channels)
+        self.out_layers_silu = nn.SiLU().to_float(self.dtype)
+
+        if is_old_ms_version():
```

Comment:
I think no need to maintain old dropout for MS<2, since this mode is not tested on MS1.x